packages/embeds/embed-react/src/index.ts

```diff
@@ -1,14 +1,20 @@
 "use client";
 
-import type { GlobalCal } from "@calcom/embed-core";
+import type { GlobalCal, GlobalCalWithoutNs } from "@calcom/embed-core";
 import EmbedSnippet from "@calcom/embed-snippet";
 
 import Cal from "./Cal";
 
-export const getCalApi = (embedJsUrl?: string): Promise<GlobalCal> =>
-  new Promise(function tryReadingFromWindow(resolve) {
-    EmbedSnippet(embedJsUrl);
-    const api = window.Cal;
+export const getCalApi = (options?: {
+  embedJsUrl?: string;
+  namespace?: string;
+}): Promise<GlobalCal | GlobalCalWithoutNs> => {
+  options = options || {};
```

Comment:
We might also support options as a string as well which is backward compatible. 🤔